quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImplTest.java

```diff
@@ -44,11 +54,220 @@ public boolean matches(Collection arg0) {
+        String editedDescription = null;
+
+        Collection updatedCollection = new Collection();
+        updatedCollection.setId(id);
+        updatedCollection.setName(editedName);
+        updatedCollection.setDescription(editedDescription);
+
+        Mockito.when(collectionRepo.save(Mockito.argThat(new ArgumentMatcher<Collection>() {
+
+            @Override
+            public boolean matches(Collection arg0) {
+                return (Objects.isNull(arg0.getName()) && Objects.isNull(arg0.getDescription()));
+            }
+
+        }))).thenReturn(updatedCollection);
+
+        Collection sameCollection = managerToTest.editCollection(id.toString(), editedName, editedDescription);
+        Assert.assertEquals(id.toString(), updatedCollection.getId().toString());
+        Assert.assertEquals(editedName, updatedCollection.getName());
+        Assert.assertEquals(editedDescription, updatedCollection.getDescription());
+
+    }
+
+    @Test
+    public void test_editCollection_blankNameAndDescription() throws Exception {
+
+        String name = COLLECTION_NAME;
+        String desc = COLLECTION_DESC;
+        Collection existingCollection = new Collection();
+        ObjectId id = new ObjectId();
+        existingCollection.setId(id);
+        existingCollection.setName(name);
+        existingCollection.setDescription(desc);
+
+        Mockito.when(collectionRepo.findById(id)).thenReturn(Optional.of(existingCollection));
+
+        String editedName = BLANK;
+        String editedDescription = BLANK;
+
+        Collection editedCollection = new Collection();
+        editedCollection.setId(id);
+        editedCollection.setName(BLANK);
+        editedCollection.setDescription(BLANK);
+
+        Mockito.when(collectionRepo.save(Mockito.argThat(new ArgumentMatcher<Collection>() {
+
+            @Override
+            public boolean matches(Collection arg0) {
+                return (arg0.getName().equals(BLANK) && arg0.getDescription().equals(BLANK));
+            }
+
+        }))).thenReturn(editedCollection);
+
+        Collection updatedCollection = managerToTest.editCollection(id.toString(), editedName, editedDescription);
+        Assert.assertEquals(id.toString(), updatedCollection.getId().toString());
+        Assert.assertEquals(editedName, updatedCollection.getName());
+        Assert.assertEquals(editedDescription, updatedCollection.getDescription());
+
+    }
+
+    @Test
+    public void test_editCollection_noCollectionFoundForId() throws Exception {
+        ObjectId objectId = new ObjectId();
+        Mockito.when(collectionRepo.findById(objectId)).thenReturn(Optional.ofNullable(null));
+
+        String editedName = EDITED_NAME;
+        String editedDescription = EDITED_DESC;
+
+        Exception e = Assert.assertThrows(Exception.class,
+                ()  -> managerToTest.editCollection(objectId.toString(), editedName, editedDescription));
+        Assert.assertEquals("Collection not found!", e.getMessage());
```

Comment:
this is not a good way to assert. the text might change but the code could still be valid. As noted above, this should throw a specific type of exception and then you just need to test that particular exception is thrown here.